gdx/src/com/badlogic/gdx/graphics/g3d/utils/CameraHelper.java

```diff
@@ -0,0 +1,215 @@
+ * Copyright 2011 See AUTHORS file.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+
+package com.badlogic.gdx.graphics.g3d.utils;
+
+import com.badlogic.gdx.graphics.Camera;
+import com.badlogic.gdx.graphics.Color;
+import com.badlogic.gdx.graphics.GL20;
+import com.badlogic.gdx.graphics.Mesh;
+import com.badlogic.gdx.graphics.VertexAttribute;
+import com.badlogic.gdx.graphics.VertexAttributes.Usage;
+import com.badlogic.gdx.graphics.g3d.Material;
+import com.badlogic.gdx.graphics.g3d.Renderable;
+import com.badlogic.gdx.graphics.g3d.RenderableProvider;
+import com.badlogic.gdx.math.Vector3;
+import com.badlogic.gdx.utils.Array;
+import com.badlogic.gdx.utils.Disposable;
+import com.badlogic.gdx.utils.Pool;
+
+/** Camera helper helps you to visualize your camera. Of course, you shouldn't use camera helper with your rendering camera.
+ * <p>
+ * How to use it:
+ * </p>
+ * <pre>
+ * // Create helper
+ * CameraHelper helper = new CameraHelper(camera);
+ * // During the rendering
+ * camera.update();
+ * helper.update();
+ * modelBatch.render(helper);
+ * </pre>
+ * @author realitix */
+public class CameraHelper implements RenderableProvider, Disposable {
+	private Camera camera;
+	private Mesh mesh;
+	private Material material = new Material();
+	private Color colorFrustum = new Color(1, 0.66f, 0, 1);
+	private Color colorCone = new Color(1, 0, 0, 1);
+	private Color colorUp = new Color(0, 0.66f, 1, 1);
+	private Color colorTarget = new Color(1, 1, 1, 1);
+	private Color colorCross = new Color(0.2f, 0.2f, 0.2f, 1);
+	private float vertices[];
+	private int id;
+	private Vector3 tmp = new Vector3();
+	private Vector3 tmp2 = new Vector3();
+
+	/** Create camera helper
+	 * @param camera */
+	public CameraHelper (Camera camera) {
+		this.camera = camera;
+		init();
+	}
+
+	private void init () {
+		short[] indices = new short[] {
+			// Near
+			0, 1, 1, 2, 2, 3, 3, 0,
+			// Far
+			4, 5, 5, 6, 6, 7, 7, 4,
+			// Sides
+			0, 4, 1, 5, 2, 6, 3, 7,
+			// Cone
+			0, 8, 1, 8, 2, 8, 3, 8,
+			// Cross near
+			9, 10, 11, 12,
+			// Cross far
+			13, 14, 15, 16,
+			// Target (position -> near -> far)
+			8, 17, 17, 18,
+			// Up triangle
+			19, 2, 2, 3, 3, 19};
+
+		int maxVertices = 20;
+		mesh = new Mesh(false, maxVertices, indices.length, new VertexAttribute(Usage.Position, 3, "a_position"),
+			new VertexAttribute(
+			Usage.ColorUnpacked, 4, "a_color"));
```

Comment:
You don't loose any quality using packed color (unless you use 16bit per channel or something). In fact it is what normally is used (RGBA8888, 8 bit per channel, 32 bits total or 16 million colors if you prefer). The reason there's `ColorUnpacked` is because java doesn't support structs, so it is easier to just pass in float, but you certainly won't loose noticeable quality by using bytes instead.